### PR TITLE
add coherent funtion into targets

### PR DESCRIPTION
**Pull request by wangzhao0217**
- add dplyr::bind_rows
- add coherent into targets

### CODE REVIEW

**Review by Robinlovelace — COMMENTED** [review]
Looks promising. There are a lot of moving parts and may make sense to merge #410 first, no rush on this and great start. Can you share a visualisation of the outputs of this code @wangzhao0217 ?

File: _targets.R [hunk 1]
```diff
@@ -1285,7 +1285,7 @@ tar_target(simplified_network, {
 tar_target(
   coherent_network, {
     cue = tar_cue(mode = "always")
-    
+    devtools::load_all()
```
Comment (Robinlovelace) [inline _targets.R:1288]: I think you can remove this. When testing locally use 

```r
tar_source()
``

**Review by Robinlovelace — APPROVED** [review]

**Review by Robinlovelace — CHANGES_REQUESTED** [review]
This error needs to be fixed:

```
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: object 'all_city_coherent_networks_75' not found
```

**Comment by wangzhao0217** [comment]
> This error needs to be fixed:
> 
> ```
> Error:
> ! Error running targets::tar_make()
>   Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
>   Debugging guide: https://books.ropensci.org/targets/debugging.html
>   How to ask for help: https://books.ropensci.org/targets/help.html
>   Last error: object 'all_city_coherent_networks_75' not found
> ```

Hi Robin, pls pull, this is not the latest

**Comment by Robinlovelace** [comment]
OK my bad. Will try with latest version.

**Comment by Robinlovelace** [comment]
Also don't have this file:

```
data-raw/combined_network_tile.geojson
```

**Review by Robinlovelace — COMMENTED** [review]
Heads-up @wangzhao0217 I'm still hitting an error:

```
▶ start target pmtiles_school
✔ skip target zones_contextual
✖ error target pmtiles_coherent
▶ end pipeline [5.12 seconds]
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: object 'all_city_coherent_networks' not found
```

**Comment by Robinlovelace** [comment]
I think

```
    for (city in names(all_city_coherent_networks)) {
``` 

Should be replaced with

```
    for (city in parameters$coherent_area) {
```

**Comment by Robinlovelace** [comment]
This would also work: 

```
    for (city in names(coherent_network)) {
```

Note @wangzhao0217 the name of the target is the only thing that targets can 'see', not the name of objects within each object.

**Comment by Robinlovelace** [comment]
Looking good @wangzhao0217 only issue now is this:

```
✖ error target coherent_network
▶ end pipeline [3.081 hours]
Warning message:
15 targets produced warnings. Run targets::tar_meta(fields = warnings, complete_only = TRUE) for the messages. 
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: Cannot open "inputdata/sg_intermediatezone_bdry_2011.gpkg"; The file doesn't seem to exist.
```

Maybe add the data as a release in inputdata if not already in there (I couldn't find it).

**Comment by wangzhao0217** [comment]
> Looking good @wangzhao0217 only issue now is this:
> 
> ```
> ✖ error target coherent_network
> ▶ end pipeline [3.081 hours]
> Warning message:
> 15 targets produced warnings. Run targets::tar_meta(fields = warnings, complete_only = TRUE) for the messages. 
> Error:
> ! Error running targets::tar_make()
>   Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
>   Debugging guide: https://books.ropensci.org/targets/debugging.html
>   How to ask for help: https://books.ropensci.org/targets/help.html
>   Last error: Cannot open "inputdata/sg_intermediatezone_bdry_2011.gpkg"; The file doesn't seem to exist.
> ```
> 
> Maybe add the data as a release in inputdata if not already in there (I couldn't find it).

Hi Robin, here is the release of all inputdata for CN: https://github.com/nptscot/npt/releases/tag/Coherent-inputdata

**Comment by wangzhao0217** [comment]
Hi Robin
There are 3 files required for coherent network:
  sg_intermediatezone_bdry_2011 = sf::st_read("inputdata/sg_intermediatezone_bdry_2011.gpkg")
  las_scotland_2023 = sf::st_read("inputdata/las_scotland_2023.geojson")
can be downloaded from https://github.com/nptscot/npt/releases/tag/boundaries-2024

  MasterMap = sf::st_read("inputdata/MasterMap_Scotland.geojson")
I have shared this file with you on Team

**Comment by Robinlovelace** [comment]
Now hitting this error after running `tar_make()`:

```
▶ start target rs_commute_balanced
✖ error target coherent_network
▶ end pipeline [17.876 seconds]
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: the condition has length > 1
```

**Comment by Robinlovelace** [comment]
OK just did 

```bash
git pull
```

now trying again.

**Review by robinlovelace-ate — APPROVED** [review]

**Review by Robinlovelace — APPROVED** [review]

**Comment by Robinlovelace** [comment]
https://github.com/nptscot/npt/assets/1825120/57311972-2dc6-4433-bc51-2df3ac66b318

**Pull request merged**